csrc/device_lower/validation.cpp

```diff
@@ -353,19 +351,32 @@ class VectorizeValidator : public OptInDispatch {
         tv,
         "\n");
 
-    if (v_id->getParallelType() == ParallelType::MisalignedVectorize) {
+    std::unordered_set<IterDomain*> dep_alloc_ids;
+    for (auto alloc : tv->getMaybeAllocationDomain()) {
+      if (validator.domains_.find(alloc) != validator.domains_.end()) {
+        dep_alloc_ids.emplace(alloc);
+      }
+    }
+
+    return {validator.vectorized_id_, dep_alloc_ids};
+  }
+
+  static void validateAllocationVectorizedId(
```

Comment:
Separated finding the relevant allocation IDs and validating them. The first part will be also extended with IdModel in #3099